app/src/main/java/app/revanced/manager/ui/component/InstallerStatusDialog.kt

```diff
@@ -0,0 +1,158 @@
+            dialogKind.dismissButton?.invoke(model)
+        },
+        icon = {
+            Icon(dialogKind.icon, null)
+        },
+        title = {
+            Text(
+                text = stringResource(dialogKind.title),
+                style = MaterialTheme.typography.headlineSmall.copy(textAlign = TextAlign.Center),
+                color = MaterialTheme.colorScheme.onSurface,
+            )
+        },
+        text = {
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.spacedBy(16.dp),
+                content = {
+                    Text(stringResource(dialogKind.contentStringResId))
+                }
+            )
+        }
+    )
+}
+
+private fun installerDialogButton(
+    @StringRes buttonStringResId: Int,
+    handler: InstallerStatusDialogButtonHandler = { },
+): InstallerStatusDialogButton = { model ->
+    TextButton(onClick = {
+        model.showInstallerStatusDialog = false
+        handler(model)
+    }) { Text(stringResource(buttonStringResId)) }
+}
+
+@FromValue("flag")
+enum class DialogKind(
+    val flag: Int,
+    val title: Int,
+    @StringRes internal val contentStringResId: Int,
+    val icon: ImageVector = Icons.Outlined.ErrorOutline,
+    val confirmButton: InstallerStatusDialogButton = installerDialogButton(R.string.ok),
+    internal val dismissButton: InstallerStatusDialogButton? = null,
+) {
+    FAILURE(
+        flag = PackageInstaller.STATUS_FAILURE,
+        title = R.string.installation_failed,
+        contentStringResId = R.string.installation_failed_description,
+    ),
+    FAILURE_ABORTED(
+        flag = PackageInstaller.STATUS_FAILURE_ABORTED,
+        title = R.string.installation_cancelled,
+        contentStringResId = R.string.installation_aborted_description,
+    ),
+    FAILURE_BLOCKED(
+        flag = PackageInstaller.STATUS_FAILURE_BLOCKED,
+        title = R.string.installation_blocked,
+        contentStringResId = R.string.installation_blocked_description,
+    ),
+    FAILURE_CONFLICT(
+        flag = PackageInstaller.STATUS_FAILURE_CONFLICT,
+        title = R.string.installation_conflict,
+        contentStringResId = R.string.installation_conflict_description,
+        confirmButton = installerDialogButton(R.string.reinstall) { vm ->
+            vm.reinstall()
+        },
+        dismissButton = installerDialogButton(R.string.cancel),
+    ),
+    FAILURE_INCOMPATIBLE(
+        flag = PackageInstaller.STATUS_FAILURE_INCOMPATIBLE,
+        title = R.string.installation_incompatible,
+        contentStringResId = R.string.installation_incompatible_description,
+    ),
+    FAILURE_INVALID(
+        flag = PackageInstaller.STATUS_FAILURE_INVALID,
+        title = R.string.installation_invalid,
+        contentStringResId = R.string.installation_invalid_description,
+        confirmButton = installerDialogButton(R.string.reinstall) { vm ->
+            vm.reinstall()
+        },
```

Comment:
Maybe use a different name since you are no longer interacting with the viewmodel directly

Reply:
What are you referring to? Should I rename vm to "model"?

Reply:
Yes